### PR TITLE
disable sensor by default,  game developer can enable it by set 'acce…

### PR DESCRIPTION
**Pull request by bofeng-song**

### CODE REVIEW

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java [hunk 1]
```diff
@@ -67,13 +69,18 @@ public CocosSensorHandler(final Context context) {
     // Getter & Setter
     // ===========================================================
     public void enable() {
-        mSensorManager.registerListener(this, mAcceleration, mSamplingPeriodUs);
-        mSensorManager.registerListener(this, mAccelerationIncludingGravity, mSamplingPeriodUs);
-        mSensorManager.registerListener(this, mGyroscope, mSamplingPeriodUs);
+        if (mEnableSensor) {
+            mHasRegisterListener = true;
```
Comment (minggo) [inline native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java]: Why need mHasRegisterListener? It seems the same effect as mEnableSensor.

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java [hunk 2]
```diff
@@ -67,13 +69,18 @@ public CocosSensorHandler(final Context context) {
...
+            mSensorManager.registerListener(this, mAcceleration, mSamplingPeriodUs);
+            mSensorManager.registerListener(this, mAccelerationIncludingGravity, mSamplingPeriodUs);
+            mSensorManager.registerListener(this, mGyroscope, mSamplingPeriodUs);
+        }
     }
 
     public void disable() {
-        this.mSensorManager.unregisterListener(this);
+        if (mHasRegisterListener) {
```
Comment (minggo) [inline native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java]: enable() is controlled by mEnableSensor, and this function is controlled by mHasRegisterListener. They are not consistent.

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java (hunk 1 above)
Comment (bofeng-song) [reply native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java]: App maybe crash, while disable sensor, but call unregisterListener. Need to be recheck.

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java (hunk 2 above)
Comment (bofeng-song) [reply native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java]: App maybe crash, while disable sensor, but call unregisterListener. Need to be recheck.

File: native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java (hunk 1 above)
Comment (minggo) [reply native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java]: But it is only set to true here. And when it is set to true, no place to set it false.

Comment (bofeng-song) [reply native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java]: The recheck result is unregisterListener can be call multi-times. I have fix it

Comment (bofeng-song) [reply native/cocos/platform/android/java/src/com/cocos/lib/CocosSensorHandler.java]: unregister implement is:

synchronized (mSensorListeners) {
            SensorEventQueue queue = mSensorListeners.get(listener);
            if (queue != null) {
                boolean result;
                if (sensor == null) {
                    result = queue.removeAllSensors();
                } else {
                    result = queue.removeSensor(sensor, true);
                }
                if (result && !queue.hasSensors()) {
                    mSensorListeners.remove(listener);
                    queue.dispose();
                }
            }
        }
    }

**Review by minggo — APPROVED** [review]

**Pull request merged**